source/lumars/state.d

```diff
@@ -2,7 +2,151 @@ module lumars.state;
 
 import bindbc.lua, taggedalgebraic, lumars;
 import taggedalgebraic : visit;
-import std.typecons : Nullable;
+import std.typecons : Nullable, isTuple;
+import std.traits : hasUDA, FieldNameTuple;
+import std.meta : AliasSeq;
+
+struct ShowInLua {}
+struct HideFromLua {}
+struct AsTable {}
+struct AsUserData {}
```

Comment:
These seem to be leftovers of some other idea you're brainstorming - they appear to be unused.